README.md

```diff
@@ -118,3 +101,171 @@ Remote debugging using :2000
+
+If you have a toolchain from other sources and find problems, check if it is an issue with your toolchain, if not open an issue or better provide a pull request with a fix.
+
+#### Black Magic Debug App
+
+ * [GCC](https://gcc.gnu.org) or Clang (Clang is not strictly officially supported)
+
+If you wish to enable support for 3rd party probes, and not only the official native hardware, you will also need:
+
+ * libusb
+ * libftdi
+ * evt
+ * libhidapi
+
+### Building
+
+The project is configured and built using the meson buildsystem, you will need to create a build directory and then configure the build depending on what you want.
+
+The project has a couple of build options you may configure, which affect the resulting binaries, this is how you will configure the firmware for you respective probe, or enable certain features.
+
+A non-exhaustive list of project options:
+ * probe: Hardware platform where the BMD firmware will run
+ * targets: Enabled debug targets
+ * debug_output: Enable debug output (for debugging the BMD stack, not debug targets)
+ * rtt_support: Enable RTT (Real Time Transfer) support
+
+You may see all available project options and valid values under  `Project options` in the output of the `meson configure` command.
+
+The first configuration commands are ran from the root of your clone of the blackmagic repository
+``` sh
+cd /path/to/repository/blackmagic
+```
+
+All following commands are ran from inside the build directory.
+
+#### Building Black Magic Debug Firmware
+
+To build the firmware you need to configure the probe hardware the firmware will run on, as well as a cross-compilation toolchain used.
+
+For convenience, a cross-file for each supported hardware probe is available which provides a sane default configuration for it.
+
+The build configuration command for the native probe may look like:
+``` sh
+meson setup build --cross-file cross-file/native.ini
+```
+
+Alternatively (for advanced users), if you wish to configure manually, for instance while writing support for a new probe, or a different toolchain, you can run:
+``` sh
+meson setup build --cross-file cross-file/arm-none-eabi.ini -Dprobe=native
+```
+
+You now should have a `build` directory from where you can build the firmware, this is also where your binary files will appear.
+
+The command `meson compile` will build the default targets:
+```sh
+cd build
+meson compile
+```
+
+Ypu should now see the resulting binaries in `build`, in this case:
+```
+blackmagic_native_firmware.bin
+blackmagic_native_firmware.elf
+```
+
+These are the binary files you will use to flash to your probe.
+
+##### region `rom' overflowed
+
+It may happen, while working with non default configurations or the project latest version from git, that the firmware does not fit in the available space for the configured probe, this could look something like:
```

Comment:
```suggestion
It may happen, while working with non default configurations or the project's latest version from Git, that the firmware does not fit in the available space for the configured probe, this could look something like:
```